parcels/compilation/codegenerator.py

```diff
@@ -421,6 +426,8 @@ def __init__(self, fieldset=None, ptype=JITParticle):
         self.fieldset = fieldset
         self.ptype = ptype
         self.field_args = collections.OrderedDict()
+        if isinstance(fieldset.U, Field) and fieldset.U.gridindexingtype == "croco" and hasattr(fieldset, "H"):
+            self.field_args["H"] = fieldset.H  # CROCO requires H field
```

Comment:
Could you elaborate on this if statement? The isinstance check for U being a field feels out of place -- do we expect cases where U isn't a field? Do we expect `hasattr(fieldset, 'H') == False` at this point of the code?

Reply:
There are cases when `fieldSet.U` is a `NestedField`, this check avoided querying the `gridindexingtype` attribute for these cases.

Reply:
Well, that test is already done here
https://github.com/OceanParcels/Parcels/blob/c376f0d2056f1bafbd65b08acef6266732a69dd3/parcels/fieldset.py#L738-L740

Reply:
No we can't, because CROCO does not have an H field if it is run in _2D mode_. The check above is only if there is a `depth` dimension. 

The alternative is that we use that same test here, but I'm not sure how easy it is to check if there is a depth dimension at this point in the codeconverter...

Reply:
Ah, I see. All good